### PR TITLE
Removes deprecated usage of `getFields(cloud, fields)`

### PR DESCRIPTION
**Pull request by kunaltyagi**
Enhances #3365 by actually removing the usage from PCL, and not just deprecating the function

### CODE REVIEW

File: filters/include/pcl/filters/impl/conditional_removal.hpp [hunk 1]
```diff
@@ -56,9 +56,7 @@ pcl::FieldComparison<PointT>::FieldComparison (
 
   // Get all fields
   std::vector<pcl::PCLPointField> point_fields;
-  // Use a dummy cloud to get the field types in a clever way
```
Comment (taketwo) [inline filters/include/pcl/filters/impl/conditional_removal.hpp]: Glad to see this "clever way" gone :)

File: filters/include/pcl/filters/impl/random_sample.hpp [hunk 2]
```diff
@@ -57,7 +57,7 @@ pcl::RandomSample<PointT>::applyFilter (PointCloud &output)
     copyPointCloud (*input_, output);
     // Get X, Y, Z fields
     std::vector<pcl::PCLPointField> fields;
-    pcl::getFields (*input_, fields);
+    pcl::getFields<PointT> (fields);
```
Comment (taketwo) [inline filters/include/pcl/filters/impl/random_sample.hpp]: Now that we have a single output parameter, this looks awkward. What about adding an overload that returns a vector?

Comment (kunaltyagi) [reply filters/include/pcl/filters/impl/random_sample.hpp]: Future PR?

**Comment by taketwo** [comment]
There are still usages of `getFieldIndex`:

```
keypoints/include/pcl/keypoints/impl/sift_keypoint.hpp:109:46: warning: 'int pcl::getFieldIndex(const pcl::PointCloud<PointT>&, const string&, std::vector<pcl::PCLPointField>&) [with PointT = pcl::PointXYZI; std::__cxx11::string = std::__cxx11::basic_string<char>]' is deprecated: use getFieldIndex<typename decltype(cloud)::PointType> () instead [-Wdeprecated-declarations]
   scale_idx_ = pcl::getFieldIndex<PointOutT> (output, "scale", out_fields_);
```

```
keypoints/include/pcl/keypoints/impl/susan.hpp:309:46: warning: 'int pcl::getFieldIndex(const pcl::PointCloud<PointT>&, const string&, std::vector<pcl::PCLPointField>&) [with PointT = pcl::PointXYZRGBL; std::__cxx11::string = std::__cxx11::basic_string<char>]' is deprecated: use getFieldIndex<typename decltype(cloud)::PointType> () instead [-Wdeprecated-declarations]
   label_idx_ = pcl::getFieldIndex<PointOutT> (output, "label", out_fields_);
```

**Pull request merged**